src/gamestate/commands.cpp

```diff
@@ -599,8 +598,8 @@ void execute_begin_factory_building_construction(sys::state& state, dcon::nation
 			}
 		}
 
-		nations::adjust_foreign_investment(state, source, state.world.state_instance_get_nation_from_state_ownership(location),
-				amount);
+		state.world.nation_get_stockpiles(source, economy::money) -= amount;
```

Comment:
this will result in double paying, since the investing nation already pays for the construction as it is being built instead of the host country